nest-forms-backend/src/ginis/ginis.service.ts

```diff
@@ -325,19 +300,39 @@ export default class GinisService {
 
     await Promise.all(
       form.files.map(async (file) => {
-        if (file.ginisUploaded) return
-
-        await this.rabbitMqClientService.publishMessageToGinisAutomation(
-          GINIS_AUTOMATION_UPLOAD_QUEUE,
-          {
-            doc_id: form.ginisDocumentId,
-            msg_id: form.id,
-            s3_path: `${process.env.MINIO_SAFE_BUCKET ?? ''}/${pospID}/${form.id}/${file.minioFileName}`,
-            filename: file.fileName,
-            file_id: file.id,
-          },
-          UPLOAD_QUEUE,
-        )
+        if (file.ginisUploaded || !form.ginisDocumentId) {
+          return
+        }
+
+        try {
+          const fileStream = await this.minioClientSubservice.download(
+            this.baConfigService.minio.buckets.safe,
+            `${pospID}/${form.id}/${file.minioFileName}`,
+          )
+
+          // sometimes ginis times-out on the first try
+          const uploadFileinfo = await this.ginisHelper.retryWithDelay(
+            async () =>
+              this.ginisApiService.uploadFile(
+                form.ginisDocumentId || '', // type safety only, guaranteed to be not null
```

Comment:
First of all, there's `!form.ginisDocumentId` above, the type should be already restricted to `string`, so this shouldn't be needed.

But, either without the check I don't think this is right.

What the code is saying:
1. Pass `ginisDocumentId`.
2. If not present pass empty string.

This is exactly the reason why we are ditching the old `ConfigService`, this is unsafe and can lead to bad behaviour.

Reply:
You can either do what you are already doing:
```
        if (file.ginisUploaded || !form.ginisDocumentId) {
          return
        }
```

Or not preferable (but needed in some cases):
`form.ginisDocumentId!` which will restrict it to `string`

Reply:
With all the checks present, it still gets unhappy with me if I don't perform yet another type safety song and dance here. I will show you the behavior on call and discuss this personally, TBH I am not sure if there is just something I am overlooking. I hate this solution too, so I would be happy to solve it some nice way.